components/ConvertForm/ManageConversion.js

```diff
@@ -1,20 +1,37 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import {
   useOpenOrder,
   useClaimOrder,
   useApprove,
   useAllowance,
+  useTransactonReceipt,
 } from 'lib/web3-contracts'
 import { bigNum } from 'lib/utils'
 import ConvertSteps from 'components/ConvertSteps/ConvertSteps'
 import processing from './assets/loader.gif'
 
-function ManageConversion({ toAnj, fromAmount, toAmount, handleReturnHome }) {
+function ManageConversion({ toAnj, fromAmount, handleReturnHome }) {
   const openOrder = useOpenOrder()
   const claimOrder = useClaimOrder()
+  const transactionReceipt = useTransactonReceipt()
   const changeAllowance = useApprove()
   const getAllowance = useAllowance()
   const [conversionSteps, setConversionSteps] = useState()
+  const [convertedTotal, setConvertedTotal] = useState(bigNum(0))
+
+  const updateConvertedValue = useCallback(
+    async hash => {
+      try {
+        const receipt = await transactionReceipt(hash)
+        const receiptValue = bigNum(receipt.logs[0].data)
```

Comment:
Ahh this is _not wrong_, but we can do a bit better knowing the ABI :).

Let's try using the [`Interface` API](https://docs.ethers.io/ethers.js/html/api-advanced.html#interface) instead.